### PR TITLE
Document sign in guards

### PR DESCRIPTION
**Pull request by derekprior**

### CODE REVIEW

**Comment by halogenandtoast** [comment]
Looks good to me.

File: lib/clearance/session_status.rb [hunk 1]
```diff
@@ -1,17 +1,24 @@
 module Clearance
+  # Indicates a user was successfully signed in, passing all {SignInGuard}s.
   class SuccessStatus
+    # Is true, indicating that the sign in was successful.
     def success?
       true
     end
   end
 
+  # Indicates a failure in the {SignInGuard} stack which prevented successfull
```
Comment (mcmire) [inline lib/clearance/session_status.rb]: successfull -> successful

File: lib/clearance/session_status.rb [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
   class FailureStatus
+    # The reason the sign in failed.
     attr_reader :failure_message
 
+    # @param [String] failure_message The reason the sign in failed.
     def initialize(failure_message)
       @failure_message = failure_message
     end
 
+    # Is false, indicating that the sign in was unsuccessful.
```
Comment (mcmire) [inline lib/clearance/session_status.rb]: Do you want to use `@return false, indicating that the sign in was unsuccessful.` or does that work?

**Comment by derekprior** [comment]
@mcmire sorry - missed your comments. Will remedy in follow up commits

**Pull request merged**